### ISSUE TITLE
Strange behaviour of the output of sendto() injection

### ISSUE THREAD

**Issue by erikbor** [riboseinc/retrace#304]
@drystone can you take a look at this please?

I have an example UDP echo server:

```C
#include <unistd.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <arpa/inet.h>
#include <sys/socket.h>

#define BUFLEN 512
#define PORT 8888

void die(char *s) {
	perror(s);
	exit(1);
}

int main(void)
{
	struct sockaddr_in si_me, si_other;

	int s, i, slen = sizeof(si_other), recv_len;
	char buf[BUFLEN];

	if ((s = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP)) == -1)
		die("socket");

	memset((char *)&si_me, 0, sizeof(si_me));

	si_me.sin_family = AF_INET;
	si_me.sin_port = htons(PORT);
	si_me.sin_addr.s_addr = htonl(INADDR_ANY);

	if (bind(s, (struct sockaddr *)&si_me, sizeof(si_me)) == -1)
		die("bind");

	while (1) {
		printf("Waiting for data...");
		fflush(stdout);

		if ((recv_len = recvfrom(s, buf, BUFLEN, 0, (struct sockaddr *)&si_other, &slen)) == -1)
			die("recvfrom()");

		printf("Received packet from %s:%d\n", inet_ntoa(si_other.sin_addr), ntohs(si_other.sin_port));
		printf("Data: %s\n", buf);

		if (sendto (s, buf, recv_len, 0, (struct sockaddr *)&si_other, slen) == -1)
			die("sendto()");
	}

	close(s);
	return(0);
}
```

Running the echo server in **terminal 1**:
```console
$ ./retrace ../echo
(14222) socket(2, 2, 17) = 3
(14222) vsnprintf("bind[0.0.0.0:8888]", 128, "bind[%s:%d]" -> "bind[0.0.0.0:8888]") = 18
(14222) bind(3 [bind[0.0.0.0:8888]], 0x7ffd3a0942a0, 16) = 0
(14222) malloc(1024) = 0x560184bf0080
(14222) printf("Waiting for data..." -> "Waiting for data...") = 19
Waiting for data...Received packet from 192.168.1.155:40437
```

Sending a string to the echo server in **terminal 2**:
```
$ echo AAAAAAAAAAAAAAAAA | nc -u 192.168.1.155 8888
AAAAAAAAAAAAAAAAA
```

Output in **terminal 1**:
```
(14222) printf("Received packet from %s:%d\n" -> "Received packet from 192.168.1.155:40437[SNIP]") = 41
Data: AAAAAAAAAAAAAAAAA
be
(14222) printf("Data: %s\n" -> "Data: AAAAAAAAAAAAAAAAA\nbe\n") = 27
(14222) sendto(3 [bind[0.0.0.0:8888]], 0x7ffd3a0942c0, 18, 0, 0x7ffd3a0942b0, 16) = 18
	0000000	 4141 4141 4141 4141 4141 4141 4141 4141 410a      | AAAAAAAAAAAAAAAAA.
(14222) printf("Waiting for data..." -> "Waiting for data...") = 19
Waiting for data...
```

Now running the echo server with the following retrace configuration:
`retrace.conf`:
```
logging-global,LOG_GROUP_NET,LOG_LEVEL_ALL
stringinject,INJECT_FORMAT_STR,sendto,4:4,1
```

**Terminal 1**:
```console
$ ./retrace -f ../retrace.conf ../echo
(14243) socket(2, 2, 17) = 3
(14243) bind(3 [bind[0.0.0.0:8888]], 0x7ffc02985740, 16) = 0
Waiting for data...
```

**Terminal 2**:
```console
$ echo AAAAAAAAAAAAAAAAA | nc -u 192.168.1.155 8888
AAAA%s%s%s%sAAAAAAAAAAAAA
```

**Terminal 1**:
```
Data: AAAAAAAAAAAAAAAAA
be
(14243) sendto(3 [bind[0.0.0.0:8888]], 0x55a140d65f90, 26, 0, 0x7ffc02985750, 16) = 26 [[redirected]] [fuzzing seed: 1507087915]
	0000000	 0000 0000 0000 0000 2573 2573 4141 4141 4141 4141 | ........%s%sAAAAAAAA
	0000020	 4141 4141 410a                                    | AAAAA.
Waiting for data...
```

## issue 1:
In the retrace output (`terminal 1`) I see `0000 0000 0000 0000 2573 2573`,  and there are missing two more `2573` `%s` which are there in the output of `nc` (`terminal 2`).

## issue 2:
I also see `0000 0000 0000 0000`, which did not appear in the first run without the string injection.

**Comment by erikbor** [comment]
The same issue is also present with `INJECT_BUF_OVERFLOW`:

```
logging-global,LOG_GROUP_NET,LOG_LEVEL_ALL
stringinject,INJECT_BUF_OVERFLOW,sendto,4:4,1
```

Terminal 1:
```console
$ echo UUUUUUUUUUUUUUUUU | nc -u 192.168.1.155 8888
```

Terminal 2:
```console
$ ./retrace -f ../retrace.conf ../echo
(14258) socket(2, 2, 17) = 3
(14258) bind(3 [bind[0.0.0.0:8888]], 0x7ffc6c7d0b70, 16) = 0
Waiting for data...Received packet from 192.168.1.155:40803
Data: UUUUUUUUUUUUUUUUU
be
(14258) sendto(3 [bind[0.0.0.0:8888]], 0x5557bcce2f90, 22, 0, 0x7ffc6c7d0b80, 16) = 22 [[redirected]] [fuzzing seed: 1507088616]
	0000000	 0000 0000 0000 0000 5555 5555 5555 5555 5555 5555 | ........UUUUUUUUUUUU
	0000020	 550a                                              | U.
Waiting for data...^C
```

And in this case we don't even see the AAAA's that are generated by `INJECT_BUF_OVERFLOW`

**Comment by drystone** [comment]
@erikbor, buffers were `free`d before logging. Fixed in #305
